src/main/java/com/algorand/algosdk/crypto/LogicsigSignature.java

```diff
@@ -48,14 +48,17 @@ public LogicsigSignature(
     ) {
         this.logic = Objects.requireNonNull(logic, "program must not be null");
         this.args = args;
-        boolean verified = false;
-        try {
-            verified = Logic.checkProgram(this.logic, this.args);
-        } catch (IOException ex) {
-            throw new IllegalArgumentException("invalid program", ex);
-        }
+        Logic.sanityCheckProgram(this.logic);
+
+        // Deprecated:
```

Comment:
@ahangsu Do you wish to retain the commented out logic or is it temporary?

Asking now because I imagine the same diff will appear in other SDKs.  Open to discussion, but from my perspective, the internal logic (like the highlighted example) can be deleted along with associated tests.

Reply:
Oh, this is just temporary, for the sake of a visual reference on some testcases, these will be deleted later on.